### PR TITLE
Permit buildPlugin jdk/jenkins/label matrix to be sparse

### PR DESCRIPTION
**Pull request by olivergondza**
This change provides an alternative way for specifying the environment for the build to run permitting to exclude combinations known not to work or not to make sense.

Thanks to @abayer for pairing while working on this.

Mostly whitespace change: https://github.com/jenkins-infra/pipeline-library/pull/76/files?w=1

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
Will try to review by Monday. Generally looks good though it may require rework of some PRs due to fatal merge conflicts. But this is ok

**Comment by olivergondza** [comment]
I have added some docs for the feature. @jenkinsci/code-reviewers, could you?

File: vars/buildPlugin.groovy [hunk 1]
```diff
@@ -174,3 +174,43 @@ def call(Map params = [:]) {
         }
     }
 }
+
+static List<Map<String, String>> getConfigurations(Map params) {
+    boolean explicit = params.containsKey("configurations")
+    boolean implicit = params.containsKey('platforms') || params.containsKey('jdkVersions') || params.containsKey('jenkinsVersions')
+
+    if (explicit && implicit) {
+        error '"configurations" option can not be used with either "platforms", "jdkVersions" or "jenkinsVersions"'
```
Comment (oleg-nenashev) [inline vars/buildPlugin.groovy]: Maybe we should support it later, but OK for now

File: README.adoc [hunk 2]
```diff
@@ -22,14 +22,26 @@ buildPlugin()
...
-  number jdk tool installed
 * `repo` (default: `null`  inherit from Multibranch) - custom Git repository to check out
 * `failFast` (default: `true`) - instruct Maven tests to fail fast
 * `platforms` (default: `['linux', 'windows']`) - Labels matching platforms to
   execute the steps against in parallel
+* `jdkVersions` (default: `[8]`) - JDK version numbers, must match a version
+    number jdk tool installed
 * `jenkinsVersions`: (default: `[null]`) - a matrix of Jenkins baseline versions to build/test against in parallel (null means default,
   only available for Maven projects)
+* `configurations`: An alternative to `platforms`, `'jdkVersions'` and `'jenkinsVersions'` arguments. Those 3 form a cartesian
```
Comment (oleg-nenashev) [inline README.adoc]: would be nice to explicitly say that the option is not compatible with platforms/jdkVersions/jenkinsVersions

**Pull request merged**